backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/backup/StopVmBackupCommand.java

```diff
@@ -130,13 +136,23 @@ private boolean isLiveBackup() {
         return vmBackup.getBackupType() == VmBackupType.Live;
     }
 
+    private boolean isHybridBackup() {
+        return vmBackup.getBackupType() == VmBackupType.Hybrid;
+    }
+
     private void updateVmBackupPhase(VmBackupPhase phase) {
         log.info("Change VM '{}' backup '{}' phase from '{}' to '{}'",
                 vmBackup.getVmId(), vmBackup.getId(), vmBackup.getPhase(), phase);
         vmBackup.setPhase(phase);
         vmBackupDao.update(vmBackup);
     }
 
+    private void updateVmBackupStopped() {
+        log.info("VM '{}' backup '{}' marked as stopped", vmBackup.getVmId(), vmBackup.getId());
+        vmBackup.setStopped(true);
+        vmBackupDao.update(vmBackup);
```

Comment:
I know it's been like that before and we didn't see an issue with that, but wouldn't it be better to introduce a stored procedure that just sets the stopped flag? the problem with this approach is that there may be a race - this command fetches the backup from the db, in the meantime the backup command updates some of its properties, and those changes are overridden here

Reply:
Good point